codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/util/Strings.kt

```diff
@@ -8,18 +8,114 @@ package software.amazon.smithy.rust.codegen.core.util
 import software.amazon.smithy.utils.CaseUtils
 import software.amazon.smithy.utils.StringUtils
 
-fun String.doubleQuote(): String = StringUtils.escapeJavaString(this, "").replace(Regex("""\\u([0-9a-f]{4})""")) { matchResult: MatchResult ->
-    "\\u{" + matchResult.groupValues[1] + "}" as CharSequence
-}
+fun String.doubleQuote(): String =
+    StringUtils.escapeJavaString(this, "").replace(Regex("""\\u([0-9a-f]{4})""")) { matchResult: MatchResult ->
+        "\\u{" + matchResult.groupValues[1] + "}" as CharSequence
+    }
 
 /**
  * Double quote a string, e.g. "abc" -> "\"abc\""
  */
 fun String.dq(): String = this.doubleQuote()
 
+private fun String.splitOnWordBoundaries(): List<String> {
+    val out = mutableListOf<String>()
+    // These are whole words but cased differently, e.g. `IPv4`, `MiB`, `GiB`, `TtL`
+    val completeWords = listOf("ipv4", "ipv6", "sigv4", "mib", "gib", "ttl")
```

Comment:
nit: Will the algorithm work correctly if we add a word to the list that has a substring of it already in the list? For example, if "sigv4a" is added to the list?

Reply:
yeah should work (but sigv4a should probably work properly already because the issue is about when you get to the capital `V`)